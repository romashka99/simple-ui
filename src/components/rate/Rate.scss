@import '../../styles/colors.scss';
@import '../../styles/main.scss';
@import '../../styles/style.scss';

.rate {
	display: flex;
	align-items: center;
	cursor: pointer;

	background-color: transparent;
	border: none;
	font-size: 18px;

	&:disabled {
		cursor: not-allowed;
		color: $color-mid-light-grey;
	}
}
