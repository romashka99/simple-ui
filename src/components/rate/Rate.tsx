import React, { useState } from 'react';

import './Rate.scss';

import StarOutline from 'components/icons/StarOutline';
import StarFill from 'components/icons/StarFill';

interface IRateProps extends React.AriaAttributes {
	count?: number;
	defaultValue?: number;
	value?: number;
	disabled?: boolean;
	className?: string;
	onChange?: (value: number) => void;
}

export default function Rate({
	count = 5,
	value,
	defaultValue = 0,
	disabled,
	className,
	onChange,
	...prop
}: IRateProps) {
	const valueRate = Number(value ? value : defaultValue);
	const classes = `rate${className ? ' ' + className : ''}`;

	const [rating, setRating] = useState<number>(valueRate);
	const [hover, setHover] = useState<number>(valueRate);

	const onClick = (index: number) => {
		if (!disabled) {
			setRating(index);
			if (onChange) onChange(index + 1);
		}
	};

	const onHover = (index: number) => {
		if (!disabled) {
			setHover(index);
		}
	};

	return (
		<button
			{...prop}
			data-testid='rate'
			className={classes}
			disabled={disabled}
		>
			{[...Array(count)].map((star, index) => (
				<span
					key={index}
					data-testid={`rate-star-${index}`}
					onClick={() => onClick(index)}
					onMouseEnter={() => onHover(index)}
					onMouseLeave={() => onHover(rating)}
				>
					{index <= ((rating && hover) || hover) ? (
						<StarFill />
					) : (
						<StarOutline />
					)}
				</span>
			))}
		</button>
	);
}
