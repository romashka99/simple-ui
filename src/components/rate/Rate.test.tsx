import { render, screen, within } from '@testing-library/react';
import React from 'react';
import Rate from '.';
import userEvent from '@testing-library/user-event';

describe('Test Input', () => {
	test('without attr', () => {
		render(<Rate />);
		const rate: HTMLButtonElement = screen.getByTestId('rate');
		expect(rate.classList.length).toBe(1);
		expect(rate.classList.contains('rate')).toEqual(true);
		expect(rate.disabled).toEqual(false);
		expect(rate).toMatchSnapshot();
	});

	test('disabled', () => {
		render(<Rate disabled />);
		const rate: HTMLButtonElement = screen.getByTestId('rate');
		expect(rate.disabled).toEqual(true);
		expect(rate).toMatchSnapshot();
	});

	test('onChange', () => {
		let value = 0;
		render(
			<Rate
				onChange={valueRate => {
					value = valueRate;
				}}
			/>
		);
		const rate: HTMLButtonElement = screen.getByTestId('rate');
		let star: HTMLSpanElement = within(rate).getByTestId('rate-star-0');
		expect(value).toEqual(0);
		userEvent.click(star);
		expect(value).toEqual(1);
		expect(rate).toMatchSnapshot();
	});
});
