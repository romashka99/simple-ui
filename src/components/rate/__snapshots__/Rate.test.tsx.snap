// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Input disabled 1`] = `
<button
  class="rate"
  data-testid="rate"
  disabled=""
>
  <span
    data-testid="rate-star-0"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.8848 17.1004L6.38821 19.9521C6.31467 19.9903 6.22413 19.9616 6.18598 19.888C6.17105 19.8592 6.16585 19.8264 6.17115 19.7944L7.18472 13.6856C7.19803 13.6054 7.17149 13.5237 7.11357 13.4666L2.70292 9.12027C2.64391 9.06212 2.64321 8.96715 2.70136 8.90814C2.72412 8.88504 2.75376 8.86994 2.78583 8.86511L8.90884 7.94135C8.98925 7.92921 9.05874 7.87873 9.09512 7.806L11.8658 2.26813C11.9029 2.19404 11.993 2.16403 12.067 2.2011C12.0961 2.21561 12.1196 2.23913 12.1341 2.26813L14.9047 7.806C14.9411 7.87873 15.0106 7.92921 15.091 7.94135L21.214 8.86511C21.296 8.87746 21.3523 8.95389 21.34 9.0358C21.3351 9.06787 21.32 9.09751 21.2969 9.12027L16.8863 13.4666C16.8284 13.5237 16.8018 13.6054 16.8151 13.6856L17.8287 19.7944C17.8423 19.8761 17.787 19.9534 17.7053 19.9669C17.6733 19.9722 17.6404 19.967 17.6117 19.9521L12.1151 17.1004C12.0429 17.063 11.957 17.063 11.8848 17.1004Z"
        fill="currentColor"
        stroke="currentColor"
        stroke-linejoin="round"
        stroke-width="1.5"
      />
    </svg>
  </span>
  <span
    data-testid="rate-star-1"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1952 1.93256L8.54006 7.23801L2.67406 8.12351C2.48166 8.15254 2.30385 8.24314 2.16727 8.38173L2.08741 8.47449C1.82152 8.82921 1.85208 9.33468 2.17662 9.65449L6.40206 13.817L5.43139 19.6717C5.39954 19.8636 5.43076 20.0607 5.52037 20.2334L5.57858 20.3304C5.83224 20.6988 6.32645 20.8291 6.73372 20.6179L12.0001 17.886L17.2664 20.6179C17.4391 20.7075 17.6362 20.7387 17.8282 20.7068L17.9384 20.6814C18.3671 20.554 18.6438 20.1243 18.5687 19.6717L17.5971 13.817L21.8235 9.65449C21.9621 9.51791 22.0527 9.3401 22.0817 9.1477L22.0916 9.03502C22.1029 8.58793 21.7797 8.19196 21.326 8.12351L15.4591 7.23801L12.8049 1.93256C12.7179 1.75855 12.5768 1.61743 12.4027 1.53037C11.9582 1.30797 11.4176 1.48804 11.1952 1.93256ZM12.0001 3.67501L14.2341 8.14159L14.2947 8.24639C14.4498 8.48059 14.6978 8.64051 14.9793 8.68297L19.9161 9.42701L16.36 12.9324L16.2791 13.0225C16.1043 13.2423 16.0288 13.5276 16.0754 13.8084L16.8921 18.733L12.4606 16.4347L12.3499 16.3855C12.0868 16.2872 11.7922 16.3036 11.5395 16.4347L7.10706 18.733L7.92473 13.8084L7.93729 13.688C7.94947 13.4073 7.84284 13.1322 7.64011 12.9324L4.08306 9.42701L9.02085 8.68297C9.34249 8.63444 9.62044 8.4325 9.76598 8.14159L12.0001 3.67501Z"
        fill="currentColor"
      />
    </svg>
  </span>
  <span
    data-testid="rate-star-2"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1952 1.93256L8.54006 7.23801L2.67406 8.12351C2.48166 8.15254 2.30385 8.24314 2.16727 8.38173L2.08741 8.47449C1.82152 8.82921 1.85208 9.33468 2.17662 9.65449L6.40206 13.817L5.43139 19.6717C5.39954 19.8636 5.43076 20.0607 5.52037 20.2334L5.57858 20.3304C5.83224 20.6988 6.32645 20.8291 6.73372 20.6179L12.0001 17.886L17.2664 20.6179C17.4391 20.7075 17.6362 20.7387 17.8282 20.7068L17.9384 20.6814C18.3671 20.554 18.6438 20.1243 18.5687 19.6717L17.5971 13.817L21.8235 9.65449C21.9621 9.51791 22.0527 9.3401 22.0817 9.1477L22.0916 9.03502C22.1029 8.58793 21.7797 8.19196 21.326 8.12351L15.4591 7.23801L12.8049 1.93256C12.7179 1.75855 12.5768 1.61743 12.4027 1.53037C11.9582 1.30797 11.4176 1.48804 11.1952 1.93256ZM12.0001 3.67501L14.2341 8.14159L14.2947 8.24639C14.4498 8.48059 14.6978 8.64051 14.9793 8.68297L19.9161 9.42701L16.36 12.9324L16.2791 13.0225C16.1043 13.2423 16.0288 13.5276 16.0754 13.8084L16.8921 18.733L12.4606 16.4347L12.3499 16.3855C12.0868 16.2872 11.7922 16.3036 11.5395 16.4347L7.10706 18.733L7.92473 13.8084L7.93729 13.688C7.94947 13.4073 7.84284 13.1322 7.64011 12.9324L4.08306 9.42701L9.02085 8.68297C9.34249 8.63444 9.62044 8.4325 9.76598 8.14159L12.0001 3.67501Z"
        fill="currentColor"
      />
    </svg>
  </span>
  <span
    data-testid="rate-star-3"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1952 1.93256L8.54006 7.23801L2.67406 8.12351C2.48166 8.15254 2.30385 8.24314 2.16727 8.38173L2.08741 8.47449C1.82152 8.82921 1.85208 9.33468 2.17662 9.65449L6.40206 13.817L5.43139 19.6717C5.39954 19.8636 5.43076 20.0607 5.52037 20.2334L5.57858 20.3304C5.83224 20.6988 6.32645 20.8291 6.73372 20.6179L12.0001 17.886L17.2664 20.6179C17.4391 20.7075 17.6362 20.7387 17.8282 20.7068L17.9384 20.6814C18.3671 20.554 18.6438 20.1243 18.5687 19.6717L17.5971 13.817L21.8235 9.65449C21.9621 9.51791 22.0527 9.3401 22.0817 9.1477L22.0916 9.03502C22.1029 8.58793 21.7797 8.19196 21.326 8.12351L15.4591 7.23801L12.8049 1.93256C12.7179 1.75855 12.5768 1.61743 12.4027 1.53037C11.9582 1.30797 11.4176 1.48804 11.1952 1.93256ZM12.0001 3.67501L14.2341 8.14159L14.2947 8.24639C14.4498 8.48059 14.6978 8.64051 14.9793 8.68297L19.9161 9.42701L16.36 12.9324L16.2791 13.0225C16.1043 13.2423 16.0288 13.5276 16.0754 13.8084L16.8921 18.733L12.4606 16.4347L12.3499 16.3855C12.0868 16.2872 11.7922 16.3036 11.5395 16.4347L7.10706 18.733L7.92473 13.8084L7.93729 13.688C7.94947 13.4073 7.84284 13.1322 7.64011 12.9324L4.08306 9.42701L9.02085 8.68297C9.34249 8.63444 9.62044 8.4325 9.76598 8.14159L12.0001 3.67501Z"
        fill="currentColor"
      />
    </svg>
  </span>
  <span
    data-testid="rate-star-4"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1952 1.93256L8.54006 7.23801L2.67406 8.12351C2.48166 8.15254 2.30385 8.24314 2.16727 8.38173L2.08741 8.47449C1.82152 8.82921 1.85208 9.33468 2.17662 9.65449L6.40206 13.817L5.43139 19.6717C5.39954 19.8636 5.43076 20.0607 5.52037 20.2334L5.57858 20.3304C5.83224 20.6988 6.32645 20.8291 6.73372 20.6179L12.0001 17.886L17.2664 20.6179C17.4391 20.7075 17.6362 20.7387 17.8282 20.7068L17.9384 20.6814C18.3671 20.554 18.6438 20.1243 18.5687 19.6717L17.5971 13.817L21.8235 9.65449C21.9621 9.51791 22.0527 9.3401 22.0817 9.1477L22.0916 9.03502C22.1029 8.58793 21.7797 8.19196 21.326 8.12351L15.4591 7.23801L12.8049 1.93256C12.7179 1.75855 12.5768 1.61743 12.4027 1.53037C11.9582 1.30797 11.4176 1.48804 11.1952 1.93256ZM12.0001 3.67501L14.2341 8.14159L14.2947 8.24639C14.4498 8.48059 14.6978 8.64051 14.9793 8.68297L19.9161 9.42701L16.36 12.9324L16.2791 13.0225C16.1043 13.2423 16.0288 13.5276 16.0754 13.8084L16.8921 18.733L12.4606 16.4347L12.3499 16.3855C12.0868 16.2872 11.7922 16.3036 11.5395 16.4347L7.10706 18.733L7.92473 13.8084L7.93729 13.688C7.94947 13.4073 7.84284 13.1322 7.64011 12.9324L4.08306 9.42701L9.02085 8.68297C9.34249 8.63444 9.62044 8.4325 9.76598 8.14159L12.0001 3.67501Z"
        fill="currentColor"
      />
    </svg>
  </span>
</button>
`;

exports[`Test Input onChange 1`] = `
<button
  class="rate"
  data-testid="rate"
>
  <span
    data-testid="rate-star-0"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.8848 17.1004L6.38821 19.9521C6.31467 19.9903 6.22413 19.9616 6.18598 19.888C6.17105 19.8592 6.16585 19.8264 6.17115 19.7944L7.18472 13.6856C7.19803 13.6054 7.17149 13.5237 7.11357 13.4666L2.70292 9.12027C2.64391 9.06212 2.64321 8.96715 2.70136 8.90814C2.72412 8.88504 2.75376 8.86994 2.78583 8.86511L8.90884 7.94135C8.98925 7.92921 9.05874 7.87873 9.09512 7.806L11.8658 2.26813C11.9029 2.19404 11.993 2.16403 12.067 2.2011C12.0961 2.21561 12.1196 2.23913 12.1341 2.26813L14.9047 7.806C14.9411 7.87873 15.0106 7.92921 15.091 7.94135L21.214 8.86511C21.296 8.87746 21.3523 8.95389 21.34 9.0358C21.3351 9.06787 21.32 9.09751 21.2969 9.12027L16.8863 13.4666C16.8284 13.5237 16.8018 13.6054 16.8151 13.6856L17.8287 19.7944C17.8423 19.8761 17.787 19.9534 17.7053 19.9669C17.6733 19.9722 17.6404 19.967 17.6117 19.9521L12.1151 17.1004C12.0429 17.063 11.957 17.063 11.8848 17.1004Z"
        fill="currentColor"
        stroke="currentColor"
        stroke-linejoin="round"
        stroke-width="1.5"
      />
    </svg>
  </span>
  <span
    data-testid="rate-star-1"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1952 1.93256L8.54006 7.23801L2.67406 8.12351C2.48166 8.15254 2.30385 8.24314 2.16727 8.38173L2.08741 8.47449C1.82152 8.82921 1.85208 9.33468 2.17662 9.65449L6.40206 13.817L5.43139 19.6717C5.39954 19.8636 5.43076 20.0607 5.52037 20.2334L5.57858 20.3304C5.83224 20.6988 6.32645 20.8291 6.73372 20.6179L12.0001 17.886L17.2664 20.6179C17.4391 20.7075 17.6362 20.7387 17.8282 20.7068L17.9384 20.6814C18.3671 20.554 18.6438 20.1243 18.5687 19.6717L17.5971 13.817L21.8235 9.65449C21.9621 9.51791 22.0527 9.3401 22.0817 9.1477L22.0916 9.03502C22.1029 8.58793 21.7797 8.19196 21.326 8.12351L15.4591 7.23801L12.8049 1.93256C12.7179 1.75855 12.5768 1.61743 12.4027 1.53037C11.9582 1.30797 11.4176 1.48804 11.1952 1.93256ZM12.0001 3.67501L14.2341 8.14159L14.2947 8.24639C14.4498 8.48059 14.6978 8.64051 14.9793 8.68297L19.9161 9.42701L16.36 12.9324L16.2791 13.0225C16.1043 13.2423 16.0288 13.5276 16.0754 13.8084L16.8921 18.733L12.4606 16.4347L12.3499 16.3855C12.0868 16.2872 11.7922 16.3036 11.5395 16.4347L7.10706 18.733L7.92473 13.8084L7.93729 13.688C7.94947 13.4073 7.84284 13.1322 7.64011 12.9324L4.08306 9.42701L9.02085 8.68297C9.34249 8.63444 9.62044 8.4325 9.76598 8.14159L12.0001 3.67501Z"
        fill="currentColor"
      />
    </svg>
  </span>
  <span
    data-testid="rate-star-2"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1952 1.93256L8.54006 7.23801L2.67406 8.12351C2.48166 8.15254 2.30385 8.24314 2.16727 8.38173L2.08741 8.47449C1.82152 8.82921 1.85208 9.33468 2.17662 9.65449L6.40206 13.817L5.43139 19.6717C5.39954 19.8636 5.43076 20.0607 5.52037 20.2334L5.57858 20.3304C5.83224 20.6988 6.32645 20.8291 6.73372 20.6179L12.0001 17.886L17.2664 20.6179C17.4391 20.7075 17.6362 20.7387 17.8282 20.7068L17.9384 20.6814C18.3671 20.554 18.6438 20.1243 18.5687 19.6717L17.5971 13.817L21.8235 9.65449C21.9621 9.51791 22.0527 9.3401 22.0817 9.1477L22.0916 9.03502C22.1029 8.58793 21.7797 8.19196 21.326 8.12351L15.4591 7.23801L12.8049 1.93256C12.7179 1.75855 12.5768 1.61743 12.4027 1.53037C11.9582 1.30797 11.4176 1.48804 11.1952 1.93256ZM12.0001 3.67501L14.2341 8.14159L14.2947 8.24639C14.4498 8.48059 14.6978 8.64051 14.9793 8.68297L19.9161 9.42701L16.36 12.9324L16.2791 13.0225C16.1043 13.2423 16.0288 13.5276 16.0754 13.8084L16.8921 18.733L12.4606 16.4347L12.3499 16.3855C12.0868 16.2872 11.7922 16.3036 11.5395 16.4347L7.10706 18.733L7.92473 13.8084L7.93729 13.688C7.94947 13.4073 7.84284 13.1322 7.64011 12.9324L4.08306 9.42701L9.02085 8.68297C9.34249 8.63444 9.62044 8.4325 9.76598 8.14159L12.0001 3.67501Z"
        fill="currentColor"
      />
    </svg>
  </span>
  <span
    data-testid="rate-star-3"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1952 1.93256L8.54006 7.23801L2.67406 8.12351C2.48166 8.15254 2.30385 8.24314 2.16727 8.38173L2.08741 8.47449C1.82152 8.82921 1.85208 9.33468 2.17662 9.65449L6.40206 13.817L5.43139 19.6717C5.39954 19.8636 5.43076 20.0607 5.52037 20.2334L5.57858 20.3304C5.83224 20.6988 6.32645 20.8291 6.73372 20.6179L12.0001 17.886L17.2664 20.6179C17.4391 20.7075 17.6362 20.7387 17.8282 20.7068L17.9384 20.6814C18.3671 20.554 18.6438 20.1243 18.5687 19.6717L17.5971 13.817L21.8235 9.65449C21.9621 9.51791 22.0527 9.3401 22.0817 9.1477L22.0916 9.03502C22.1029 8.58793 21.7797 8.19196 21.326 8.12351L15.4591 7.23801L12.8049 1.93256C12.7179 1.75855 12.5768 1.61743 12.4027 1.53037C11.9582 1.30797 11.4176 1.48804 11.1952 1.93256ZM12.0001 3.67501L14.2341 8.14159L14.2947 8.24639C14.4498 8.48059 14.6978 8.64051 14.9793 8.68297L19.9161 9.42701L16.36 12.9324L16.2791 13.0225C16.1043 13.2423 16.0288 13.5276 16.0754 13.8084L16.8921 18.733L12.4606 16.4347L12.3499 16.3855C12.0868 16.2872 11.7922 16.3036 11.5395 16.4347L7.10706 18.733L7.92473 13.8084L7.93729 13.688C7.94947 13.4073 7.84284 13.1322 7.64011 12.9324L4.08306 9.42701L9.02085 8.68297C9.34249 8.63444 9.62044 8.4325 9.76598 8.14159L12.0001 3.67501Z"
        fill="currentColor"
      />
    </svg>
  </span>
  <span
    data-testid="rate-star-4"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1952 1.93256L8.54006 7.23801L2.67406 8.12351C2.48166 8.15254 2.30385 8.24314 2.16727 8.38173L2.08741 8.47449C1.82152 8.82921 1.85208 9.33468 2.17662 9.65449L6.40206 13.817L5.43139 19.6717C5.39954 19.8636 5.43076 20.0607 5.52037 20.2334L5.57858 20.3304C5.83224 20.6988 6.32645 20.8291 6.73372 20.6179L12.0001 17.886L17.2664 20.6179C17.4391 20.7075 17.6362 20.7387 17.8282 20.7068L17.9384 20.6814C18.3671 20.554 18.6438 20.1243 18.5687 19.6717L17.5971 13.817L21.8235 9.65449C21.9621 9.51791 22.0527 9.3401 22.0817 9.1477L22.0916 9.03502C22.1029 8.58793 21.7797 8.19196 21.326 8.12351L15.4591 7.23801L12.8049 1.93256C12.7179 1.75855 12.5768 1.61743 12.4027 1.53037C11.9582 1.30797 11.4176 1.48804 11.1952 1.93256ZM12.0001 3.67501L14.2341 8.14159L14.2947 8.24639C14.4498 8.48059 14.6978 8.64051 14.9793 8.68297L19.9161 9.42701L16.36 12.9324L16.2791 13.0225C16.1043 13.2423 16.0288 13.5276 16.0754 13.8084L16.8921 18.733L12.4606 16.4347L12.3499 16.3855C12.0868 16.2872 11.7922 16.3036 11.5395 16.4347L7.10706 18.733L7.92473 13.8084L7.93729 13.688C7.94947 13.4073 7.84284 13.1322 7.64011 12.9324L4.08306 9.42701L9.02085 8.68297C9.34249 8.63444 9.62044 8.4325 9.76598 8.14159L12.0001 3.67501Z"
        fill="currentColor"
      />
    </svg>
  </span>
</button>
`;

exports[`Test Input without attr 1`] = `
<button
  class="rate"
  data-testid="rate"
>
  <span
    data-testid="rate-star-0"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.8848 17.1004L6.38821 19.9521C6.31467 19.9903 6.22413 19.9616 6.18598 19.888C6.17105 19.8592 6.16585 19.8264 6.17115 19.7944L7.18472 13.6856C7.19803 13.6054 7.17149 13.5237 7.11357 13.4666L2.70292 9.12027C2.64391 9.06212 2.64321 8.96715 2.70136 8.90814C2.72412 8.88504 2.75376 8.86994 2.78583 8.86511L8.90884 7.94135C8.98925 7.92921 9.05874 7.87873 9.09512 7.806L11.8658 2.26813C11.9029 2.19404 11.993 2.16403 12.067 2.2011C12.0961 2.21561 12.1196 2.23913 12.1341 2.26813L14.9047 7.806C14.9411 7.87873 15.0106 7.92921 15.091 7.94135L21.214 8.86511C21.296 8.87746 21.3523 8.95389 21.34 9.0358C21.3351 9.06787 21.32 9.09751 21.2969 9.12027L16.8863 13.4666C16.8284 13.5237 16.8018 13.6054 16.8151 13.6856L17.8287 19.7944C17.8423 19.8761 17.787 19.9534 17.7053 19.9669C17.6733 19.9722 17.6404 19.967 17.6117 19.9521L12.1151 17.1004C12.0429 17.063 11.957 17.063 11.8848 17.1004Z"
        fill="currentColor"
        stroke="currentColor"
        stroke-linejoin="round"
        stroke-width="1.5"
      />
    </svg>
  </span>
  <span
    data-testid="rate-star-1"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1952 1.93256L8.54006 7.23801L2.67406 8.12351C2.48166 8.15254 2.30385 8.24314 2.16727 8.38173L2.08741 8.47449C1.82152 8.82921 1.85208 9.33468 2.17662 9.65449L6.40206 13.817L5.43139 19.6717C5.39954 19.8636 5.43076 20.0607 5.52037 20.2334L5.57858 20.3304C5.83224 20.6988 6.32645 20.8291 6.73372 20.6179L12.0001 17.886L17.2664 20.6179C17.4391 20.7075 17.6362 20.7387 17.8282 20.7068L17.9384 20.6814C18.3671 20.554 18.6438 20.1243 18.5687 19.6717L17.5971 13.817L21.8235 9.65449C21.9621 9.51791 22.0527 9.3401 22.0817 9.1477L22.0916 9.03502C22.1029 8.58793 21.7797 8.19196 21.326 8.12351L15.4591 7.23801L12.8049 1.93256C12.7179 1.75855 12.5768 1.61743 12.4027 1.53037C11.9582 1.30797 11.4176 1.48804 11.1952 1.93256ZM12.0001 3.67501L14.2341 8.14159L14.2947 8.24639C14.4498 8.48059 14.6978 8.64051 14.9793 8.68297L19.9161 9.42701L16.36 12.9324L16.2791 13.0225C16.1043 13.2423 16.0288 13.5276 16.0754 13.8084L16.8921 18.733L12.4606 16.4347L12.3499 16.3855C12.0868 16.2872 11.7922 16.3036 11.5395 16.4347L7.10706 18.733L7.92473 13.8084L7.93729 13.688C7.94947 13.4073 7.84284 13.1322 7.64011 12.9324L4.08306 9.42701L9.02085 8.68297C9.34249 8.63444 9.62044 8.4325 9.76598 8.14159L12.0001 3.67501Z"
        fill="currentColor"
      />
    </svg>
  </span>
  <span
    data-testid="rate-star-2"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1952 1.93256L8.54006 7.23801L2.67406 8.12351C2.48166 8.15254 2.30385 8.24314 2.16727 8.38173L2.08741 8.47449C1.82152 8.82921 1.85208 9.33468 2.17662 9.65449L6.40206 13.817L5.43139 19.6717C5.39954 19.8636 5.43076 20.0607 5.52037 20.2334L5.57858 20.3304C5.83224 20.6988 6.32645 20.8291 6.73372 20.6179L12.0001 17.886L17.2664 20.6179C17.4391 20.7075 17.6362 20.7387 17.8282 20.7068L17.9384 20.6814C18.3671 20.554 18.6438 20.1243 18.5687 19.6717L17.5971 13.817L21.8235 9.65449C21.9621 9.51791 22.0527 9.3401 22.0817 9.1477L22.0916 9.03502C22.1029 8.58793 21.7797 8.19196 21.326 8.12351L15.4591 7.23801L12.8049 1.93256C12.7179 1.75855 12.5768 1.61743 12.4027 1.53037C11.9582 1.30797 11.4176 1.48804 11.1952 1.93256ZM12.0001 3.67501L14.2341 8.14159L14.2947 8.24639C14.4498 8.48059 14.6978 8.64051 14.9793 8.68297L19.9161 9.42701L16.36 12.9324L16.2791 13.0225C16.1043 13.2423 16.0288 13.5276 16.0754 13.8084L16.8921 18.733L12.4606 16.4347L12.3499 16.3855C12.0868 16.2872 11.7922 16.3036 11.5395 16.4347L7.10706 18.733L7.92473 13.8084L7.93729 13.688C7.94947 13.4073 7.84284 13.1322 7.64011 12.9324L4.08306 9.42701L9.02085 8.68297C9.34249 8.63444 9.62044 8.4325 9.76598 8.14159L12.0001 3.67501Z"
        fill="currentColor"
      />
    </svg>
  </span>
  <span
    data-testid="rate-star-3"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1952 1.93256L8.54006 7.23801L2.67406 8.12351C2.48166 8.15254 2.30385 8.24314 2.16727 8.38173L2.08741 8.47449C1.82152 8.82921 1.85208 9.33468 2.17662 9.65449L6.40206 13.817L5.43139 19.6717C5.39954 19.8636 5.43076 20.0607 5.52037 20.2334L5.57858 20.3304C5.83224 20.6988 6.32645 20.8291 6.73372 20.6179L12.0001 17.886L17.2664 20.6179C17.4391 20.7075 17.6362 20.7387 17.8282 20.7068L17.9384 20.6814C18.3671 20.554 18.6438 20.1243 18.5687 19.6717L17.5971 13.817L21.8235 9.65449C21.9621 9.51791 22.0527 9.3401 22.0817 9.1477L22.0916 9.03502C22.1029 8.58793 21.7797 8.19196 21.326 8.12351L15.4591 7.23801L12.8049 1.93256C12.7179 1.75855 12.5768 1.61743 12.4027 1.53037C11.9582 1.30797 11.4176 1.48804 11.1952 1.93256ZM12.0001 3.67501L14.2341 8.14159L14.2947 8.24639C14.4498 8.48059 14.6978 8.64051 14.9793 8.68297L19.9161 9.42701L16.36 12.9324L16.2791 13.0225C16.1043 13.2423 16.0288 13.5276 16.0754 13.8084L16.8921 18.733L12.4606 16.4347L12.3499 16.3855C12.0868 16.2872 11.7922 16.3036 11.5395 16.4347L7.10706 18.733L7.92473 13.8084L7.93729 13.688C7.94947 13.4073 7.84284 13.1322 7.64011 12.9324L4.08306 9.42701L9.02085 8.68297C9.34249 8.63444 9.62044 8.4325 9.76598 8.14159L12.0001 3.67501Z"
        fill="currentColor"
      />
    </svg>
  </span>
  <span
    data-testid="rate-star-4"
  >
    <svg
      fill="currentColor"
      height="1em"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M11.1952 1.93256L8.54006 7.23801L2.67406 8.12351C2.48166 8.15254 2.30385 8.24314 2.16727 8.38173L2.08741 8.47449C1.82152 8.82921 1.85208 9.33468 2.17662 9.65449L6.40206 13.817L5.43139 19.6717C5.39954 19.8636 5.43076 20.0607 5.52037 20.2334L5.57858 20.3304C5.83224 20.6988 6.32645 20.8291 6.73372 20.6179L12.0001 17.886L17.2664 20.6179C17.4391 20.7075 17.6362 20.7387 17.8282 20.7068L17.9384 20.6814C18.3671 20.554 18.6438 20.1243 18.5687 19.6717L17.5971 13.817L21.8235 9.65449C21.9621 9.51791 22.0527 9.3401 22.0817 9.1477L22.0916 9.03502C22.1029 8.58793 21.7797 8.19196 21.326 8.12351L15.4591 7.23801L12.8049 1.93256C12.7179 1.75855 12.5768 1.61743 12.4027 1.53037C11.9582 1.30797 11.4176 1.48804 11.1952 1.93256ZM12.0001 3.67501L14.2341 8.14159L14.2947 8.24639C14.4498 8.48059 14.6978 8.64051 14.9793 8.68297L19.9161 9.42701L16.36 12.9324L16.2791 13.0225C16.1043 13.2423 16.0288 13.5276 16.0754 13.8084L16.8921 18.733L12.4606 16.4347L12.3499 16.3855C12.0868 16.2872 11.7922 16.3036 11.5395 16.4347L7.10706 18.733L7.92473 13.8084L7.93729 13.688C7.94947 13.4073 7.84284 13.1322 7.64011 12.9324L4.08306 9.42701L9.02085 8.68297C9.34249 8.63444 9.62044 8.4325 9.76598 8.14159L12.0001 3.67501Z"
        fill="currentColor"
      />
    </svg>
  </span>
</button>
`;
